rpcserver.go

```diff
@@ -6049,6 +6049,7 @@ func (r *rpcServer) AddInvoice(ctx context.Context,
 		NumHops:                  globalBlindCfg.NumHops,
```

Comment:
Nit: Maybe something like this in the commit msg body:

This commit makes sure that when an incoming channel list is supplied for the blinded route, that it changes the other blinding restrictions accordingly. Moreover it introduces a sanity check when the number of paths for the blinded route is set to 0.